### ISSUE TITLE
Develop memory strategy for large input files

### ISSUE THREAD

**Issue by slz4025** [rainflame/geopolygonize#25]
Currently, we read the entire TIF file and split it into tiles. If the file is very big, it may not be possible to load the whole file into memory. Investigate a way to determine the pixel size and metadata of the file and get regions of the file from disk.

**Comment by slz4025** [comment]
In addressing issue #48 with PR #<TODO>, we store in-memory data for the following:
1. original input TIF,
2. cleaned raster,
3. polygonized raster (shapefile).

We can potentially fetch data by tile from disk for cleaning, but likely not for vectorizing as the input is now a shapefile.

Consider examining these techniques for the rasters: https://stackoverflow.com/questions/14351255/techniques-for-working-with-large-numpy-arrays

**Comment by slz4025** [comment]
After #51 and thinking about this more, I think a good goal is to _minimize_ how often we need to load a large amount of data into memory.

We can use the following strategy:
1. Input the data and break it into tiles, where each tile is saved to file.
2. For each step--clean, polygonize, vectorize--operate per tile and create intermediate files.
3. Have a last step to stitch all the files together and make the output file.

With this, we only need to have 2 steps that might use a substantial amount of memory. The rest of the flow can use a bounded amount of memory on the order of number of processes * tile size * tile size.

**Comment by slz4025** [comment]
Numpy supports `mmap_mode` when loading a file, which allows the data to be accessed as if in memory, though it is actually in disk. 
https://numpy.org/doc/stable/reference/generated/numpy.load.html
